StorageUtility/Linux/Device.cpp

```diff
@@ -15,90 +15,53 @@ See the License for the specific language governing permissions and
 limitations under the License.
 </License>
 */
+
+#include <string.h>
+#include <errno.h>
+
 #include "Device.h"
 #include "StorageUtility.h"
 
 namespace vtStor
 {
 
-/*
-cDevice::cDevice(const PSP_DEVINFO_DATA& DevInfoData, const PSP_DEVICE_INTERFACE_DATA& DevInterfaceData, const PSP_INTERFACE_DEVICE_DETAIL_DATA& DevDetailData, U32 SizeOfDevDetailData)
-{
-    AllocateMemories(DevInfoData, DevInterfaceData, DevDetailData, SizeOfDevDetailData);
-    CopyMemories(DevInfoData, DevInterfaceData, DevDetailData, SizeOfDevDetailData);
-}
-*/
-
-/*cDevice::~cDevice()
+cDevice::cDevice()
 {
-    DeallocateMemories();
-}*/
 
-/*void cDevice::Data(std::unordered_map<eDeviceDataType, void*>& Data)
-{
-  //  Data.insert({ eDeviceDataType::PSP_DEVINFO_DATA, m_DevInfoData });
-   // Data.insert({ eDeviceDataType::PSP_DEVICE_INTERFACE_DATA, m_DevInterfaceData });
-   // Data.insert({ eDeviceDataType::PSP_INTERFACE_DEVICE_DETAIL_DATA, m_DevDetailData });
-}*/
+}
 
-/*DeviceHandle cDevice::Handle()
+cDevice::~cDevice()
 {
-    eErrorCode errorCode;
-    DeviceHandle deviceHandle;
 
-    errorCode = GetStorageDeviceHandle(m_DevDetailData->DevicePath, deviceHandle);
-    if (eErrorCode::None != errorCode)
-    {
-        throw std::exception("Failed to create device handle", GetLastError());
-    }
-    return deviceHandle;
-}*/
-/*
+}
 
-eErrorCode cDevice::GetStorageDeviceHandle(const String& DevicePath, HANDLE& Handle)
+cDevice::cDevice (String &Path)

```

Comment:
What if this Path variable is deleted? So, a device initialized will lost m_DevicePath value.